lib/collection/src/recommendations.rs

```diff
@@ -290,15 +290,15 @@ fn recommend_by_avg_vector(
 
     let lookup_collection_name = lookup_from.as_ref().map(|x| &x.collection);
 
-    let positive_vectors = convert_to_vectors(
-        positive.iter(),
+    let positive_vectors = convert_to_vectors_owned(
```

Comment:
It affects regular dense vector and do unnecessary copying. We can fix this bug by changing `combine_aggregate` and allow to do this function with any order of indices. In that case we can avoid any changes in this file

Reply:
Which exact copying are you referring to?
Here I am passing ownership of the vector and consuming it.

Reply:
`convert_to_vectors` returns `impl Iterator<Item = VectorRef<'a>>` without ownership and we use it in dense case for avg calculation with allocation for result vector only. Here I see a copying of each positive vector in dense case where it's not required